src/main/cpp/tests/cast_string.cpp

```diff
@@ -229,4 +231,212 @@ TYPED_TEST(StringToIntegerTests, Overflow)
   }();
 
   CUDF_TEST_EXPECT_COLUMNS_EQUAL(result->view(), expected);
-}
\ No newline at end of file
+}
+
+TEST_F(StringToDecimalTests, Simple)
+{
+  auto const strings = test::strings_column_wrapper({"1", "0", "-1"});
+  strings_column_view scv{strings};
+
+  auto const result =
+    spark_rapids_jni::string_to_decimal(1, 0, scv, false, rmm::cuda_stream_default);
+
+  test::fixed_point_column_wrapper<int32_t> expected({1, 0, -1}, {1, 1, 1}, numeric::scale_type{0});
+
+  test::print(result->view());
+
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(result->view(), expected);
+}
+
+TEST_F(StringToDecimalTests, OverPrecise)
+{
+  auto const strings = test::strings_column_wrapper({"123456", "999999", "-123456", "-999999"});
+  strings_column_view scv{strings};
+
+  auto const result =
+    spark_rapids_jni::string_to_decimal(5, 0, scv, false, rmm::cuda_stream_default);
+
+  test::fixed_point_column_wrapper<int32_t> expected(
+    {12346, 100000, -12346, -100000}, {0, 0, 0, 0}, numeric::scale_type{0});
+
+  test::print(result->view());
+
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(result->view(), expected);
+}
+
+TEST_F(StringToDecimalTests, Rounding)
+{
+  auto const strings = test::strings_column_wrapper({"1.23456", "9.99999", "-1.23456", "-9.99999"});
+  strings_column_view scv{strings};
+
+  auto const result =
+    spark_rapids_jni::string_to_decimal(5, 4, scv, false, rmm::cuda_stream_default);
+
+  test::fixed_point_column_wrapper<int32_t> expected(
+    {12346, 100000, -12346, -100000}, {1, 1, 1, 1}, numeric::scale_type{-4});
```

Comment:
So it is expected that the java code still needs to do an overflow check on the output of this? The second and fourth numbers technically do not fit in a precision of 5. The precision is 6 after rounding. If this is expected that is fine, we just need to make sure that we have documented it. Just because it is not what I would have expected.

Reply:
Not at all. My goal is to match `spark-shell` in the handling. I thought I was doing that, but I see that spark will return null for these tests. I will update the code to ensure it matches the output of spark.

Reply:
Updated these to match spark-shell and updated the code to produce valid results.